packages/peregrine/src/List/list.js

```diff
@@ -16,7 +16,10 @@ class List extends Component {
             .isRequired,
         renderItem: PropTypes.oneOfType([PropTypes.func, PropTypes.string]),
         onSelectionChange: PropTypes.func,
-        selectionModel: PropTypes.oneOf(['checkbox', 'radio'])
+        selectionModel: PropTypes.oneOf(['checkbox', 'radio']),
+        isLoading: PropTypes.bool,
+        renderLoadingState: PropTypes.func,
+        renderEmptyState: PropTypes.func
```

Comment:
I understand the intent here, but I would prefer to see a single render prop for the list, with an API that is similar to [Apollo's `Query`](https://www.apollographql.com/docs/react/essentials/queries.html).

```es6
({ loading, empty, ... }) => { ... }
```

Reply:
Yeah, I understood, but in this case there will be no sense to pass loading and empty properties to render function, Apollo's Query stores loading state and query result internally, in our case we pass isLoading to List component as a property, the same for the items, we pass items as a property to List. The idea was to reduce boilerplate and avoid if...else statements. When we use Query we always start our render function with such checks and it is a boilerplate actually.

```
<Query>
{({ loading, error }) => {
if (loading) {
  return 'Loading...';
}

if (error) {
  return 'Error';
}
}}
</Query>
```.

It is OK if you do not like the idea of pull request, but there is no sense to pass loading, empty to render function, in this case we will just pass this property through List to render function without using them, we can calculate this properties directly in the render function. If you do not like the idea, then we can close this pull request.

Reply:
@zetlen Could you share your opinion with us?